paseos/attitude/disturbance_calculations.py

```diff
@@ -12,11 +12,34 @@ def calculate_aero_torque():
     return np.array(T)
 
 
-def calculate_grav_torque():
-    # calculations for torques
-    # T must be in actor body fixed frame (to be discussed)
-    T = [0, 0, 0]
-    return np.array(T)
+def calculate_grav_torque(u_r, u_n, J, r):
+    """
+    Equation for gravity gradient torque with up to J2 effect from:
+    https://doi.org/10.1016/j.asr.2018.06.025, chapter 3.3
+    This function currently only works for Earth centered orbits
+
+    Args:
+        u_r (np.array): Unit vector pointing from Satellite center of gravity to Earth's center of gravity
+        u_n (np.array): Unit vector along the Earth's rotation axis, in the spacecraft body frame
+        J (np.array): The satellites moment of inertia, in the form of [[Ixx Ixy Ixz]
+                                                                        [Iyx Iyy Iyx]
+                                                                        [Izx Izy Izz]]
+        h (float): The distance from the center of the Earth to the satellite
+
+    Returns:
+        np.array: total gravitational torques in Nm expressed in the spacecraft body frame
+    """
+    # Constants
+    mu = 3.986004418e14  # Earth's gravitational parameter, [m^3/s^2]
```

Comment:
Please do not define these constants here. Those are properties of the central body. You should import the central body as part of the input and get those constants. Otherwise, they will be redefined multiple times.

Reply:
Have changed this in the code